### DISCUSSION TITLE
How do I test a secured API endpoint?

### DISCUSSION THREAD

**Discussion by benaissazaki** [nextauthjs/next-auth#1527] [Help] [answered]
I have this simple API and I want to test it:
```
export default async (req, res) => {  
   const session = async getSession({ req });
   if(session){
     // do things
   } else {
      // error response
   }
}
```

I heard of node-mocks-http to mock requests and responses, but how can I mock a `req` so that `session` evaluates to true during the test?

**Thank you**

**Comment by balazsorban44** [comment, marked as answer]
Maybe you could mock getSession itself, or even better, mock the request to /api/auth/session with

https://mswjs.io/

https://kentcdodds.com/blog/stop-mocking-fetch

**Reply by benaissazaki** [reply]
Thank you, I am a beginner in testing so I didn't understand very well how mocking works, but after some researches I just mocked getSession as you suggested.
